test/unit/BPool/BPool_ExitPool.tree

```diff
@@ -0,0 +1,25 @@
+BPool::ExitPool
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when total supply is zero
+│   └── it should revert // division by zero
+├── when pool amount in is too small
+│   └── it should revert
+├── when caller does not have the required pool shares
+│   └── it should revert
+└── when caller has enough pool shares
+    ├── when balance of pool in any token is zero
+    │   └── it should revert
+    ├── when returned amount of a token is less than minAmountsOut
+    │   └── it should revert
+    └── when preconditions are met
+        ├── it emits LOG_CALL event
+        ├── it sets the reentrancy lock
+        ├── it pulls poolAmountIn shares
+        ├── it sends zero exitFee to factory // note: should we remove this altogether?
```

Comment:
i'd leave it skipped perhaps, or we can do calcs with `EXIT_FEE` being zero, but that should pass if we change it to a value (import it from BConst?)

Reply:
I like tests to also cover (meaning: break on) meaningfully different configurations. If we were to, later down the line, set up the `EXIT_FEE` to be something else, failing tests would force us to review the behaviour changes induced by it, plus having the dedicated branch in the tree makes it more evident that we're doing something fishy (if having a configurable zero/non-zero exit fee was a real feature, we'd skip the call if it was zero, and it'd be it's own branch)

Reply:
why would we have failing tests if we properly test it?
let's say `EXIT_FEE` is 1%, tests should expect `amount * EXIT_FEE` to be pushed to factory, since it is now zero, we expect a (non-hardcoded) zero amount being pushed